src/components/modal/tenants/AddEditTenantDialog.tsx

```diff
@@ -159,35 +206,58 @@ export default class AddTenantManuallyDialog extends React.Component<Props, Stat
             />
           )}
         />
-        {showSubdomainAlreadyUsedError && (
-          <Text style={style.inputError}>{I18n.t('general.subdomainAlreadyUsed', { tenantName: tenantNameWithSameSubdomain })}</Text>
+        {errorSubdomainAlreadyUsed && (
+          <Text style={style.inputError}>{I18n.t('general.subdomainAlreadyUsed', { tenantName: tenantNameWithSameSubdomain.name })}</Text>
         )}
       </View>
     );
   }
 
+  private async editTenant(newSubdomain: string, newName: string, newEndpointCloud: EndpointCloud) {
+    const { tenantIndex, tenants } = this.props;
+    let formIsValid = Utils.validateInput(this, this.formCreateTenantValidationDef);
+    if (this.tenant?.subdomain !== newSubdomain) {
+      const foundTenant = tenants.find((tenantConnection) => tenantConnection.subdomain === newSubdomain);
+      if (foundTenant) {
+        formIsValid = false;
+        this.setState({ tenantNameWithSameSubdomain: foundTenant, errorSubdomainAlreadyUsed: true });
+      }
+    }
+    if (formIsValid) {
+      const editedTenant: TenantConnection = {
+        subdomain: newSubdomain,
+        name: newName,
+        endpoint: newEndpointCloud?.endpoint
+      };
+      tenants[tenantIndex] = editedTenant;
```

Comment:
[low] You could have put the Json doc in the var without creating editedTenant.
I think I already told you that before, right ?

Reply:
This var is used two times

Reply:
My bad then!